kairon/shared/llm/gpt3.py

```diff
@@ -52,10 +53,8 @@ async def train(self, *args, **kwargs) -> Dict:
             await self.__create_collection__(collection)
             for content in tqdm(collections['content'], desc="Training FAQ"):
                 if content['content_type'] == CognitionDataType.json.value:
-                    if not content['metadata'] or []:
-                        search_payload, vector_embeddings = content['data'], json.dumps(content['data'])
-                    else:
-                        search_payload, vector_embeddings = Utility.get_embeddings_and_payload_data(content['data'], content['metadata'])
+                    metadata = processor.find_matching_metadata(self.bot, content['data'], content.get('collection'))
+                    search_payload, vector_embeddings = processor.get_embeddings_and_payload_data(content['data'], metadata)
```

Comment:
this is supposed to be a utility
either move to Utility or create static method within Processor

Reply:
also rename to retrieve_search_payload_and_embedding_payload()